src/main/java/org/elasticsearch/index/mapper/core/ShortFieldMapper.java

```diff
@@ -209,14 +208,11 @@ public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower
     }
 
     @Override
-    public Filter nullValueFilter() {
+    public Query nullValueFilter() {
         if (nullValue == null) {
             return null;
         }
-        return new QueryWrapperFilter(NumericRangeQuery.newIntRange(names.indexName(), precisionStep,
-                nullValue.intValue(),
-                nullValue.intValue(),
-                true, true));
+        return new ConstantScoreQuery(termQuery(nullValue, null));
```

Comment:
How come nullValueFilter (no idea what this method is doing, its name is horrible, are we sure we really need it??????) is CSQ(term()) here for short/float but NRQ for long. Why the inconsistency?